### PR TITLE
[JBIDE-22059] make sure route chooser "no route error" has a shell

### PR DESCRIPTION
**Pull request by adietish**

### CODE REVIEW

**Comment by adietish** [comment]
@xcoulon can you please review this small patch?

**Comment by jcantrill** [comment]
LGTM

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/route/RouteChooser.java [hunk 1]
```diff
@@ -41,11 +41,7 @@ public RouteChooser(Shell shell) {
 	public IRoute chooseRoute(List<IRoute> routes) {
 		final IRoute[] selectedRoute = new IRoute[1];
 		Display.getDefault().syncExec(() -> {
-			if (shell == null) {
```
Comment (xcoulon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/route/RouteChooser.java]: what's the point in having a `shell` field in this class ? Why not just call this `PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell()` when a shell is needed, instead ?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/route/RouteChooser.java [hunk 2]
```diff
@@ -55,6 +51,14 @@ public IRoute chooseRoute(List<IRoute> routes) {
...
 	}
 
+	private void ensureHasShell() {
+		if (shell == null) {
+			shell = PlatformUI.getWorkbench()
+					.getActiveWorkbenchWindow()
+					.getShell();
+		}
+	}
+
```
Comment (xcoulon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/route/RouteChooser.java]: from comment above, this method could return the active workbench window's shell, to be used when necessary.

Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/route/RouteChooser.java]: +1 you're absolutely right.

**Pull request merged**